src/fetch/block_number.py

```diff
@@ -0,0 +1,44 @@
+"""
+Fetches the closest block number for a given timestamp from Etherscan API
+Ref: https://docs.etherscan.io/api-endpoints/blocks#get-block-number-by-timestamp
+"""
+import json
+import os
+import time
+from datetime import datetime
+from enum import Enum
+
+import requests
+from dotenv import load_dotenv
+
+ETHERSCAN_API_URL = "https://api.etherscan.io/api"
+
+
+class Closest(Enum):
+    """
+    Meant for telling Etherscan API whether you want the
+    first block before or after a given timestamp.
+    """
+
+    BEFORE = "before"
+    AFTER = "after"
+
+
+def get_block(when: datetime, closest: Closest) -> int:
+    """Fetches ethereum mainnet `closest` block number for a given datetime object"""
+    epoch = int(time.mktime(when.timetuple()))
+    load_dotenv()
+    response = requests.get(
+        url=ETHERSCAN_API_URL,
+        params=json.dumps(
+            {
+                "module": "block",
+                "action": "getblocknobytime",
+                "apikey": os.environ["ETHERSCAN_API_KEY"],
+                "timestamp": epoch,
+                "closest": str(closest.value),
```

Comment:
So this is an IDE complaint. EnumS don't have to have string values (they could be numbers). In this case since the json isn't actually typed we don't technically need it (it's just a habit because I don't like when pycharm complains.

Reply:
Just forgot.

Reply:
But but how do you expect the str method to be called if I don't use str(obj)?

Reply:
In this particular case the json isn't typed anyway so there is no need to cast to string.